### PR TITLE
[Feature] Adds --ssh_username to tools/launch.py

### PR DESCRIPTION
**Pull request by erickim555**
## Description
This PR adds the `--ssh_username` flag to the distributed launcher `tools/launch.py`.

Context: `tools/launch.py` issues remote commands via the `ssh` command, in the following way:
```
# suppose the remote host is {ip}, listening at {port}, and we want to run {cmd} on the remote host
ssh -o StrictHostKeyChecking=no -p {port} {ip} '{cmd}'
```
By default, this `ssh` command will use the username inherited by the caller. However, sometimes it's necessary to explicitly change the ssh username to avoid SSH permission errors, eg `ssh bob@1.2.3.4` vs `ssh 1.2.3.4`.

This PR allows the user to optionally pass in an `--ssh_username=bob` so that the above ssh command turns into:
```
# note: the "bob@" prefix to {ip}!
ssh -o StrictHostKeyChecking=no -p {port} bob@{ip} '{cmd}'
```

From the CLI, the usage is:
```
# Launch distributed code, using username=bob
python3.7 tools/launch.py \
--workspace {workspace_dir_nfs} \
...
--ssh_username bob
```

If not provided, `--ssh_username` falls back to the empty string `""`, which reduces to the original behavior (no username prefix to the `{ip}`).

Testing: On my end, I performed an end-to-end test, and verified that this PR works when using `tools/launch.py` to launch a simple "hello world"-type job via `tools/launch.py ... --ssh_username erickim` on a small two machine cluster.

Addresses: https://github.com/dmlc/dgl/issues/3159

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Review by VoVAllen — APPROVED** [review]
Thanks for the contribution! Looks good to me

**Comment by erickim555** [comment]
Thank you for taking a look!
@VoVAllen  I don't have write access, so I can't merge this into master. What's the correct procedure here? Can you grant me write access? Or will you merge it into master on my behalf? 
Thanks!

**Comment by VoVAllen** [comment]
Thanks! I just merged

**Pull request merged**
